app/docker/networks/edit/NetworkDetailsView.tsx

```diff
@@ -0,0 +1,138 @@
+  const deleteNetworkMutation = useDeleteNetwork();
+  const filters = {
+    network: [networkId],
+  };
+  const containersQuery = useContainers(environmentId, filters);
+
+  useEffect(() => {
+    if (networkQuery.data && containersQuery.data) {
+      setFilteredNetworkContainers(
+        filterContainersInNetwork(networkQuery.data, containersQuery.data)
+      );
+    }
+  }, [networkQuery.data, containersQuery.data]);
+
+  return (
+    <>
+      <PageHeader
+        title="Network details"
+        breadcrumbs={[
+          { link: 'docker.networks', label: 'Networks' },
+          {
+            link: 'docker.networks.network',
+            label: networkQuery.data?.Name || '',
+          },
+        ]}
+      />
+      {networkQuery.data && (
+        <>
+          <NetworkDetailsTable
+            network={networkQuery.data}
+            onRemoveNetworkClicked={onRemoveNetworkClicked}
+          />
+
+          <AccessControlPanel
+            onUpdateSuccess={() =>
+              queryClient.invalidateQueries([
+                'environments',
+                environmentId,
+                'docker',
+                'networks',
+                networkId,
+              ])
+            }
+            resourceControl={networkQuery.data.Portainer?.ResourceControl}
+            resourceType={ResourceControlType.Network}
+            disableOwnershipChange={isSystemNetwork(networkQuery.data.Name)}
+            resourceId={networkId}
+          />
+        </>
+      )}
+      {networkQuery.data?.Options && (
+        <NetworkOptionsTable options={networkQuery.data.Options} />
+      )}
+      {filteredNetworkContainers && (
+        <NetworkContainersTable
+          networkContainers={filteredNetworkContainers}
+          nodeName={nodeName}
+          environmentId={environmentId}
+          networkId={networkId}
+        />
+      )}
+    </>
+  );
+
+  async function onRemoveNetworkClicked() {
+    const message = 'Do you want to delete the network?';
+    const confirmed = await confirmDeletionAsync(message);
+
+    if (confirmed) {
+      deleteNetworkMutation.mutate(
+        { environmentId, networkId },
+        {
+          onSuccess: () => {
+            router.stateService.go('docker.networks');
+          },
+        }
+      );
+    }
+  }
+
+  function filterContainersInNetwork(
+    network: DockerNetwork,
+    containers: DockerContainer[]
+  ) {
+    const containersInNetwork: NetworkContainer[] = [];
+    containers.forEach((container) => {
+      if (network.Containers) {
+        const containerInNetworkResponse: NetworkResponseContainer =
+          network.Containers[container.Id];
+        if (containerInNetworkResponse) {
+          const containerInNetwork: NetworkContainer = {
+            ...containerInNetworkResponse,
+            Id: container.Id,
+          };
+          containersInNetwork.push(containerInNetwork);
+        }
+      }
+    });
+    return containersInNetwork;
```

Comment:
```suggestion
    if (!network.Containers) {
       return []
     }
    return _.compact(containers.map(container => {
       const containerInNetworkResponse: NetworkResponseContainer =
          network.Containers[container.Id];
        if (containerInNetworkResponse) {
          const containerInNetwork: NetworkContainer = {
            ...containerInNetworkResponse,
            Id: container.Id,
          };
          return containerInNetwork
        }
    }))
```

Reply:
Thankyou for the suggestion! 
Typescript didn't pick up on the !network.Containers check inside the map function. To avoid another if statement in the map function, I checked the docker api to see if the Containers field in DockerNetwork always exists (even if it's an empty object).
https://docs.docker.com/engine/api/v1.24/#inspect-network
https://docs.docker.com/engine/api/v1.41/#operation/NetworkInspect

Since Containers is always returned, I updated the type definition to make Containers always mandatory. I also updated other fields to be mandatory